compiler/src/dotty/tools/dotc/core/Designators.scala

```diff
@@ -23,8 +26,49 @@ object Designators {
 
     def asTerm(implicit ctx: Context): TermDesignator = unsupported("asTerm")
     def asType(implicit ctx: Context): TypeDesignator = unsupported("asType")
+
+    def withNameSpace(space: NameSpace)(implicit ctx: Context): Designator { type ThisName = self.ThisName } =
```

Comment:
Shouldn't this function be defined in `Name` instead of `Designator`?

Reply:
No, because it is overridden in LocalName.